tests/check_check_sub.c

```diff
@@ -497,6 +497,337 @@ START_TEST(test_ck_assert_uint_expr)
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ne(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ne_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ne(1%d, 1%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_lt)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.0;
+  double y = 2.5;
+  ck_assert_floating_lt(x, y);
+  y-=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_lt(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_lt_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_lt(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.0;
+  double y = 2.5;
+  ck_assert_floating_le(x, y);
+  ck_assert_floating_le(x, x);
+  y-=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_gt)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.5;
+  double y = 2.0;
+  ck_assert_floating_gt(x, y);
+  y+=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_gt(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_gt_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_gt(2%d, 3%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.5;
+  double y = 2.0;
+  ck_assert_floating_ge(x, y);
+  ck_assert_floating_ge(x, x);
+  y+=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge(2%d, 3%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_tol10)
+{
+  record_test_name(tcase_name());
+
+  double x = 0.0001;
+  double y = 0.0002;
+  ck_assert_floating_eq_tol10(x, y, -3);
+  ck_assert_floating_eq_tol10(x, x, -3);
+  x*=10;
+  y*=10;
+  ck_assert_floating_eq_tol10(x, y, -2);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq_tol10(x, y, -3);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_tol10_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq_tol10(3%d, 2%f, -1);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_tol10_min_tol10)
+{
+  record_test_name(tcase_name());
+
+  int min_tol10 = LDBL_MIN_10_EXP;
+  ck_assert_floating_eq_tol10(0, 0, min_tol10);
+  // TODO: test with LDBL_MAX_10_EXP+1 and platform specific output.
```

Comment:
Oh, I see. As you do not know what the maximum or minimum number is offline, and certainly the value may be different between different platforms, you cannot know what the failure message will be.